scipy/stats/tests/test_mstats_basic.py

```diff
@@ -420,6 +420,17 @@ def test_mode(self):
         assert_equal(mstats.mode(ma4, axis=0), ([[3,2]], [[1,1]]))
         assert_equal(mstats.mode(ma4, axis=-1), ([[2],[3],[5]], [[1],[1],[1]]))
 
+        a1_res = mstats.mode(a1, axis=None)
+        assert_equal(str(a1_res)[:4], 'Mode')
+
+        # test namedtuple attributes by index
+        attributes = ('mode', 'count')
+        for i, attr in enumerate(attributes):
+            assert_equal(a1_res[i], getattr(a1_res, attr))
+
+        # test namedtuple attributes explicitly by name
+        assert_equal(a1_res.mode, getattr(a1_res, 'mode'))
+        assert_equal(a1_res.count, getattr(a1_res, 'count'))
```

Comment:
Yes. This test is redundant; I wanted to make the `a1_res.mode` call explicit (in addition to `a1_res[0]`) but if this creates more confusion than clarity it should be scrapped.

I didn't do it in a different function for consistency within this module. I agree that there should be a class and the tests should be separated by functions but that would require rewriting the rest of the test module if we wanted to keep things consistent; I think that is out of scope for this PR (note that in the `test_stats` module, the tests are indeed done as you mentioned).